eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/producer/EventMeshProducer.java

```diff
@@ -83,32 +86,24 @@ public synchronized void init(EventMeshHTTPConfiguration eventMeshHttpConfigurat
         keyValue.put("eventMeshIDC", eventMeshHttpConfiguration.getEventMeshIDC());
         mqProducerWrapper = new MQProducerWrapper(eventMeshHttpConfiguration.getEventMeshConnectorPluginType());
         mqProducerWrapper.init(keyValue);
-        inited.compareAndSet(false, true);
         log.info("EventMeshProducer [{}] inited.............", producerGroupConfig.getGroupName());
     }
 
 
-    public synchronized void start() throws Exception {
-        if (started.get()) {
+    public void start() throws Exception {
+
+        if (!started.compareAndSet(false, true)) {
             return;
         }
-
         mqProducerWrapper.start();
-        started.compareAndSet(false, true);
         log.info("EventMeshProducer [{}] started.............", producerGroupConfig.getGroupName());
     }
 
-    public synchronized void shutdown() throws Exception {
-        if (!inited.get()) {
-            return;
-        }
-
-        if (!started.get()) {
+    public void shutdown() throws Exception {
+        if (!inited.compareAndSet(true, false && !started.compareAndSet(true, false))) {
```

Comment:
the code logic of this paragraph is easy to make people confused, may be you use `if(!inited.compareAndSet(true, false) {...}`, `if(!started.compareAndSet(true, false) {...}` much better.

Reply:
@xwm1992  I will fix it and resubmit this pr later